armi/utils/mathematics.py

```diff
@@ -290,6 +291,39 @@ def getStepsFromValues(values, prevValue=0.0):
     return steps
 
 
+def isMonotonic(inputList, relation):
+    """
+    Checks if an iterable contains elements that are monotonically increasing or
+    decreasing, whatever that might mean for the specific types of the elements.
+
+    Parameters
+    ----------
+    inputList : list
+        Some list to check. Values in the list should have a defined relation to
+        each other.
+    relation : {'increaseOrEqual', 'increaseExclusive', 'decreaseOrEqual', 'decreaseExclusive'}
+        The relation between the elements to check.
+
+    Returns
+    -------
+    bool
+    """
+    try:
+        op = {
+            "increaseOrEqual": operator.le,
```

Comment:
I _think_ you only need two of these.

For instance, if you are testing "less than or equal to", you don't need to also test "strictly less then".

I _think_ you can replace these for with `>=` and `<=`.